saleor/graphql/attribute/utils.py

```diff
@@ -750,15 +784,43 @@ def _pre_save_file_value(
             instance, attribute, "file_url", attr_value.file_url
         )
         if value is None:
-            value = attribute_models.AttributeValue(
+            value = AttributeValue(
                 attribute=attribute,
                 file_url=file_url,
                 name=name,
                 content_type=attr_value.content_type,
             )
             value.slug = generate_unique_slug(value, name)
-            value.save()
-        return (value,)
+            return ((AttributeValue.objects.create, value),)
+        return ((None, value),)
+
+    @classmethod
+    def _prepare_attribute_values(cls, attribute, values):
+        results, values_to_create = prepare_attribute_values(attribute, values)
+        return [(None, record) for record in results] + [
+            (AttributeValue.objects.create, record) for record in values_to_create
+        ]
+
+    @classmethod
+    def _bulk_create_pre_save_values(cls, pre_save_bulk):
+        results: Dict["Attribute", list[AttributeValue]] = defaultdict(list)
+
+        for action, attribute_data in pre_save_bulk.items():
+            for attribute, values in attribute_data.items():
+                if action == AttributeValue.objects.create:
```

Comment:
Is there any reason why we use the manager method as an action type? Maybe some kind of enum would be better for that?